### PR TITLE
bluetooth: Fix duration endianness issues

### PR DESCRIPTION
**Pull request by Tronil**
Added missing endianness conversion in controller and fixed double
conversion in host

Signed-off-by: Troels Nilsson <trnn@demant.com>

### CODE REVIEW

File: subsys/bluetooth/host/adv.c [hunk 1]
```diff
@@ -351,7 +351,7 @@ int bt_le_adv_set_enable_ext(struct bt_le_ext_adv *adv,
 	net_buf_add_u8(buf, 1);
 
 	net_buf_add_u8(buf, adv->handle);
-	net_buf_add_le16(buf, param ? sys_cpu_to_le16(param->timeout) : 0);
+	net_buf_add_le16(buf, param ? param->timeout : 0);
```
Comment (alwa-nordic) [inline subsys/bluetooth/host/adv.c:354]: This is incorrect. `net_buf_add_le16` takes a native integer.

**Review by alwa-nordic — CHANGES_REQUESTED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Comment by alwa-nordic** [comment]
Looks ok on the host part. This should ideally be in two separate PRs, because you need different people to review the different files.

**Review by wopu-ot — APPROVED** [review]
LGTM; I prefer having this in a single PR to keep host and controller in sync.

**Review by cvinayak — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
Same PR is fine, but different commits would be better.

**Pull request merged**